<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Ghislain Bidaut, Plateforme Cibi, CRCM, Aix-Marseille Université" />


<title>TD Analyse de données Single Cell RNA-seq sous R - Package Seurat</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/simplex.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Formation Single Cell RNA-seq sous R</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Accueil</a>
</li>
<li>
  <a href="Introduction_Formation.html">Introduction</a>
</li>
<li>
  <a href="Introduction_R_RStudio.html">Introduction R/Rstudio</a>
</li>
<li>
  <a href="Analyse_scRNA-seq_Seurat.html">Analyse scRNA-seq sous Seurat</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">TD Analyse de données Single Cell RNA-seq
sous R - Package Seurat</h1>
<h4 class="author">Ghislain Bidaut, Plateforme Cibi, CRCM, Aix-Marseille
Université</h4>
<h4 class="date">Novembre 2025</h4>

</div>


<p><br> <br></p>
<p><img src="Maxi_Logo_Formation_Bioinfo.png" width="3000" /></p>
<p><br> <br> <br> <br></p>
<div id="introduction" class="section level1" number="1">
<h1><span class="header-section-number">1</span> Introduction</h1>
<p>Dans ce TD, nous allons faire une <strong>analyse de données Single
Cell</strong> sous <strong>R</strong>.</p>
<p>Ces données sont présentées dans la publication par Steele <em>et
al</em>: Multimodal mapping of the tumor and peripheral blood immune
landscape in human pancreatic cancer. <em>Nature Cancer</em> 1, pp
1097–1112 (2020).</p>
<p><strong>L’objectif de l’analyse</strong> est <strong>d’identifier le
paysage immunitaire dans les adénocarcinomes ductal pancréatiques
(PDAC)</strong>.</p>
<p>Nous allons utiliser <strong>Seurat</strong> pour les étapes
d’analyse suivantes:</p>
<ul>
<li>Chargement et contrôle qualité d’un échantillon</li>
<li>Extraction des clusters et visualisation UMAP</li>
<li>Interpretation des clusters - analyse interactive</li>
<li>Extraction des gènes différentiellement exprimés - visualistion de
clusters</li>
</ul>
</div>
<div id="données" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Données</h1>
<p>Les données relatives à cette publication sont disponibles sur le
site de <strong>GEO</strong> sous le numéro d’accession
<strong>GSE155698</strong>: <a
href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE155698"
class="uri">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE155698</a></p>
<div id="tableau-récapitulatif-des-données" class="section level2"
number="2.1">
<h2><span class="header-section-number">2.1</span> Tableau récapitulatif
des données</h2>
<table>
<thead>
<tr class="header">
<th align="left">GEO</th>
<th align="right">Sample</th>
<th align="right">Type</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GSM4710689</td>
<td align="right">PDAC_TISSUE_1</td>
<td align="right">PDAC</td>
</tr>
<tr class="even">
<td align="left">GSM4710690</td>
<td align="right">PDAC_TISSUE_2</td>
<td align="right">PDAC</td>
</tr>
<tr class="odd">
<td align="left">GSM4710691</td>
<td align="right">PDAC_TISSUE_3</td>
<td align="right">PDAC</td>
</tr>
<tr class="even">
<td align="left">GSM4710692</td>
<td align="right">PDAC_TISSUE_4</td>
<td align="right">PDAC</td>
</tr>
<tr class="odd">
<td align="left">GSM4710693</td>
<td align="right">PDAC_TISSUE_5</td>
<td align="right">PDAC</td>
</tr>
<tr class="even">
<td align="left">GSM4710694</td>
<td align="right">PDAC_TISSUE_6</td>
<td align="right">PDAC</td>
</tr>
<tr class="odd">
<td align="left">GSM4710695</td>
<td align="right">PDAC_TISSUE_7</td>
<td align="right">PDAC</td>
</tr>
<tr class="even">
<td align="left">GSM4710696</td>
<td align="right">PDAC_TISSUE_8</td>
<td align="right">PDAC</td>
</tr>
<tr class="odd">
<td align="left">GSM4710697</td>
<td align="right">PDAC_TISSUE_9</td>
<td align="right">PDAC</td>
</tr>
<tr class="even">
<td align="left">GSM4710698</td>
<td align="right">PDAC_TISSUE_10</td>
<td align="right">PDAC</td>
</tr>
<tr class="odd">
<td align="left">GSM4710699</td>
<td align="right">PDAC_TISSUE_11A</td>
<td align="right">PDAC</td>
</tr>
<tr class="even">
<td align="left">GSM4710700</td>
<td align="right">PDAC_TISSUE_11B</td>
<td align="right">PDAC</td>
</tr>
<tr class="odd">
<td align="left">GSM4710701</td>
<td align="right">PDAC_TISSUE_12</td>
<td align="right">PDAC</td>
</tr>
<tr class="even">
<td align="left">GSM4710702</td>
<td align="right">PDAC_TISSUE_13</td>
<td align="right">PDAC</td>
</tr>
<tr class="odd">
<td align="left">GSM4710704</td>
<td align="right">PDAC_TISSUE_15</td>
<td align="right">PDAC</td>
</tr>
<tr class="even">
<td align="left">GSM4710706</td>
<td align="right">AdjNorm_TISSUE_1</td>
<td align="right">Norm</td>
</tr>
<tr class="odd">
<td align="left">GSM4710707</td>
<td align="right">AdjNorm_TISSUE_2</td>
<td align="right">Norm</td>
</tr>
<tr class="even">
<td align="left">GSM4710708</td>
<td align="right">AdjNorm_TISSUE_3</td>
<td align="right">Norm</td>
</tr>
</tbody>
</table>
</div>
<div id="environment" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Environment</h2>
<p>Charger la librairie <code>Seurat</code>.</p>
<p>Importance du <code>set.seed()</code> pour les étapes stochastiques
(PCA, clustering)!</p>
<pre class="r"><code>#install.packages(&quot;Seurat&quot;)

library(Seurat)</code></pre>
<pre><code>## Loading required package: SeuratObject</code></pre>
<pre><code>## Loading required package: sp</code></pre>
<pre><code>## &#39;SeuratObject&#39; was built under R 4.5.0 but the current version is
## 4.5.1; it is recomended that you reinstall &#39;SeuratObject&#39; as the ABI
## for R may have changed</code></pre>
<pre><code>## &#39;SeuratObject&#39; was built with package &#39;Matrix&#39; 1.7.3 but the current
## version is 1.7.4; it is recomended that you reinstall &#39;SeuratObject&#39; as
## the ABI for &#39;Matrix&#39; may have changed</code></pre>
<pre><code>## 
## Attaching package: &#39;SeuratObject&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, t</code></pre>
<pre class="r"><code>set.seed(1234) # pour reproductibilité</code></pre>
</div>
<div id="récupération-des-données-de-comptage" class="section level2"
number="2.3">
<h2><span class="header-section-number">2.3</span> Récupération des
données de comptage</h2>
<p>Les données de Steele sont disponible sur le chemin:
<code>/shared/projects/tp_2529_ngs_amu_181496/data/GSE155698_RAW"</code>.</p>
</div>
</div>
<div id="analyse-de-léchantillon-pdac-1" class="section level1"
number="3">
<h1><span class="header-section-number">3</span> Analyse de
l’échantillon PDAC 1</h1>
<p>Dans cette première partie, nous allons analyse uniquemeent
l’échantillon <strong>PDAC_TISSUE_1</strong>.</p>
<div id="charger-les-données-de-léchantillon-pdac-1-avec-seurat."
class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Charger les données
de l’échantillon PDAC 1 avec Seurat.</h2>
<ol style="list-style-type: decimal">
<li>Créer un objet Seurat et garder</li>
</ol>
<ul>
<li>les cellules ayant au <strong>moins 200 gènes exprimés</strong></li>
<li>les gènes exprimés au moins <strong>dans 3 cellules</strong></li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Examiner la matrice de comptage: Examiner les gènes
<code>NKG7</code>, <code>KRT18</code> et <code>PRSS1</code> ainsi que le
marqueur <code>CD45</code> sur les 30 premières cellules.</li>
</ol>
<pre class="r"><code># Chemin des données d&#39;entrée

# input_dir_pdac &lt;- &quot;/shared/projects/tp_2529_ngs_amu_181496/data/GSE155698_RAW/PDAC_TISSUE_1/filtered_feature_bc_matrix/&quot; 

input_dir_pdac &lt;- &quot;GSE155698_RAW/PDAC_TISSUE_1/filtered_feature_bc_matrix/&quot;

# Charger les données avec Read10X
pdac.data &lt;- Read10X(data.dir = input_dir_pdac)

# Créer l&#39;objet Seurat
pdac.seurat &lt;- CreateSeuratObject(counts = pdac.data, project = &quot;PDAC&quot;, min.cells = 3, min.features = 200)

# Voir la matrice de comptage
pdac.data[c(&quot;PTPRC&quot;, &quot;NKG7&quot;, &quot;KRT18&quot;, &quot;PRSS1&quot;), 1:30]</code></pre>
<pre><code>## 4 x 30 sparse Matrix of class &quot;dgCMatrix&quot;</code></pre>
<pre><code>##   [[ suppressing 30 column names &#39;AAACGAAAGTGGAAAG-1&#39;, &#39;AAACGAAGTAGGGTAC-1&#39;, &#39;AAACGAAGTCATAGTC-1&#39; ... ]]</code></pre>
<pre><code>##                                                                              
## PTPRC .  1   .   . 2 5 6   .   . . .    . 1 5 . .   .  . 15   .  . 12 10 .  .
## NKG7  .  .   .   . 1 1 3   1   . . .    . . . . .   .  .  .   1  . 10  . .  .
## KRT18 . 20 322   . . . 2 118 203 1 .    . . 2 . .   . 13  . 529 92  .  . 1 80
## PRSS1 .  .   . 185 . . .   .   . . . 1133 . . . . 217  .  .   .  .  .  . .  .
##                   
## PTPRC  4 1 .   . 9
## NKG7  41 . .   . 3
## KRT18  . . .   . .
## PRSS1  . . . 256 .</code></pre>
<pre class="r"><code># Taille de l&#39;objet Seurat
pdac.seurat</code></pre>
<pre><code>## An object of class Seurat 
## 19273 features across 1410 samples within 1 assay 
## Active assay: RNA (19273 features, 0 variable features)
##  1 layer present: counts</code></pre>
</div>
<div id="faire-un-contrôle-qualité" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Faire un contrôle
qualité</h2>
<p>Seurat permet de faire un contrôle qualité (QC) sur</p>
<ul>
<li>Le <strong>nombre de gènes par cellules</strong>: un nombre de gènes
aberrant peut être un marqueur de “doublets” ou de gouttes vides.</li>
<li>Le <strong>nombre de molécules uniques</strong> associée à une
cellule</li>
<li>Le poucentage de gènes associés avec la
<strong>mitochondrie</strong>.</li>
</ul>
<ol style="list-style-type: decimal">
<li>Afficher les distributions de ces 3 valeurs</li>
<li>Fixer les seuils acceptables sur ces distributions</li>
<li>Les appliquer aux données.</li>
</ol>
<pre class="r"><code># L&#39;opérateur [[, ]] permet d&quot;&#39;ajouter des données d&#39;annotation au meta data
pdac.seurat[[&quot;percent.mt&quot;]] &lt;- PercentageFeatureSet(pdac.seurat, pattern = &quot;^MT-&quot;)
VlnPlot(pdac.seurat, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent.mt&quot;), ncol = 3)</code></pre>
<p><img src="TD_scRNA-seq_Seurat_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<pre class="r"><code>pdac.seurat &lt;- subset(pdac.seurat, subset = nFeature_RNA &gt; 200 &amp; nFeature_RNA &lt; 5000 &amp; percent.mt &lt; 15)
VlnPlot(pdac.seurat, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent.mt&quot;), ncol = 3)</code></pre>
<p><img src="TD_scRNA-seq_Seurat_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
<pre class="r"><code>pdac.seurat</code></pre>
<pre><code>## An object of class Seurat 
## 19273 features across 928 samples within 1 assay 
## Active assay: RNA (19273 features, 0 variable features)
##  1 layer present: counts</code></pre>
</div>
<div id="normalisation" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> Normalisation</h2>
<p><strong>Normalisez les données</strong> avec la méthode
<em>LogNormalize</em>, en multipliant les valeurs par un facteur 10000
et en prenant le log du résultat.</p>
<div id="solution" class="section level3" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> Solution</h3>
<pre class="r"><code>pdac.seurat &lt;- NormalizeData(pdac.seurat, normalization.method = &quot;LogNormalize&quot;, scale.factor = 10000)</code></pre>
<pre><code>## Normalizing layer: counts</code></pre>
<pre class="r"><code>pdac.seurat</code></pre>
<pre><code>## An object of class Seurat 
## 19273 features across 928 samples within 1 assay 
## Active assay: RNA (19273 features, 0 variable features)
##  2 layers present: counts, data</code></pre>
</div>
</div>
<div id="préparation-acp-étape-1-identification-des-gènes-variables"
class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Préparation ACP étape
1: identification des gènes variables</h2>
<p><strong>Objectif:</strong> Il faut identifier les gènes présentant la
plus grande <strong>variation</strong> entre cellules.</p>
<ol style="list-style-type: decimal">
<li>Utiliser la fonction <code>FindVariableFeatures</code> avec la
méthode “vst” en partiant de 2000 “features” (gènes).</li>
<li>Faire une représentation des gènes les plus variables en identifiant
les 10 gènes les plus variables en les labellisant sur le
graphique.</li>
</ol>
<div id="solution-1" class="section level3" number="3.4.1">
<h3><span class="header-section-number">3.4.1</span> Solution</h3>
<pre class="r"><code>pdac.seurat &lt;- FindVariableFeatures(pdac.seurat, selection.method = &quot;vst&quot;, nfeatures = 2000)</code></pre>
<pre><code>## Finding variable features for layer counts</code></pre>
<pre class="r"><code>top10 &lt;- head(VariableFeatures(pdac.seurat), 10)

plot &lt;- VariableFeaturePlot(pdac.seurat)
plot &lt;- LabelPoints(plot = plot, points = top10, repel = TRUE)</code></pre>
<pre><code>## When using repel, set xnudge and ynudge to 0 for optimal results</code></pre>
<pre class="r"><code>plot</code></pre>
<p><img src="TD_scRNA-seq_Seurat_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<pre class="r"><code>pdac.seurat</code></pre>
<pre><code>## An object of class Seurat 
## 19273 features across 928 samples within 1 assay 
## Active assay: RNA (19273 features, 2000 variable features)
##  2 layers present: counts, data</code></pre>
</div>
</div>
<div id="préparation-acp-étape-2-mise-à-léchelle-des-données"
class="section level2" number="3.5">
<h2><span class="header-section-number">3.5</span> Préparation ACP étape
2: Mise à l’échelle des données</h2>
<p>Nous avons besoin de <strong>standardiser</strong>:</p>
<ol style="list-style-type: decimal">
<li>centrer et réduire l’expression de chaque gène dans la matrice de
comptage avec <code>ScaleData</code>.</li>
</ol>
<pre class="r"><code>pdac.seurat &lt;- ScaleData(pdac.seurat) </code></pre>
<pre><code>## Centering and scaling data matrix</code></pre>
<pre class="r"><code>pdac.seurat</code></pre>
<pre><code>## An object of class Seurat 
## 19273 features across 928 samples within 1 assay 
## Active assay: RNA (19273 features, 2000 variable features)
##  3 layers present: counts, data, scale.data</code></pre>
</div>
<div id="réduction-de-dimensions-avec-acp" class="section level2"
number="3.6">
<h2><span class="header-section-number">3.6</span> Réduction de
dimensions avec ACP</h2>
<ol style="list-style-type: decimal">
<li><strong>Faire une réduction de dimensions</strong> par ACP sur les
gènes variables (fonction <code>VariableFeatures</code>)</li>
<li><strong>Afficher</strong> les 5 plus fortes contributions pour les 5
composants principaux</li>
<li>Faire une représentation des <em>loadings</em> des 2 composants
principaux (fonction <code>VizDimLoadings</code>)</li>
<li>Faire une représentation des données dans l’espace
<strong>PC_1</strong> vs <strong>PC_2</strong></li>
<li>Faire une représentation des données sous forme de Heatmap dans
l’espace des composants principaux grâce à <code>DimHeatmap</code></li>
</ol>
<div id="solution-2" class="section level3" number="3.6.1">
<h3><span class="header-section-number">3.6.1</span> Solution</h3>
<pre class="r"><code>pdac.seurat &lt;- RunPCA(pdac.seurat, features = VariableFeatures(object = pdac.seurat))</code></pre>
<pre><code>## PC_ 1 
## Positive:  CTSB, CD68, TGFBI, GPNMB, MAFB, CTSL, CD163, GRN, LRP1, PSAP 
##     APOC1, FPR3, APOE, MSR1, NPC2, MS4A4A, HNMT, SCARB2, CD86, LHFPL2 
##     RNASE1, GPX1, LGMN, CREG1, CTSZ, MMP14, CTSH, CST3, SIGLEC1, DAB2 
## Negative:  CD7, CD247, CTSW, PRF1, CCL5, GZMA, SAMD3, IL2RB, HOPX, IL32 
##     AC092580.4, CCND2, TMIGD2, GZMB, LTB, KLRC1, TNFRSF18, NKG7, CST7, KLRC2 
##     GNLY, FASLG, KLRD1, KLRB1, TIGIT, GZMM, CD3D, NDFIP2, GPR171, CD3G 
## PC_ 2 
## Positive:  SMIM22, C19orf33, GPRC5A, PMEPA1, PDZK1IP1, LAMC2, IGFBP3, CXADR, ERBB3, SERINC2 
##     LINC01133, KRT8, CAMK2N1, BACE2, KRT19, S100A14, ITGB4, S100A16, PTRF, TEAD1 
##     DEFB1, IFI27, MALL, MAL2, KRT7, SOX9, ADIRF, CFI, TRIM29, PITX1 
## Negative:  FTL, AIF1, NCF2, S100A8, LST1, SPI1, LYZ, MNDA, BCL2A1, CTSS 
##     FCGR2A, SLC11A1, SLC2A3, CD14, KCTD12, C5AR1, FGL2, AQP9, MARCH1, IL1B 
##     S100A9, PLXDC2, PLEK, CLEC7A, OGFRL1, CYBB, TNFSF13B, MS4A7, MS4A6A, CCR1 
## PC_ 3 
## Positive:  CD7, CTSW, GZMA, CCND2, CCL5, IL2RB, HOPX, CD247, PRF1, AC092580.4 
##     GZMB, TNFRSF18, TMIGD2, NKG7, KLRC2, KLRC1, NDFIP2, LGALS1, ID3, SPRY1 
##     FASLG, ITGA1, SAMD3, PPDPF, RGS1, ACP5, GNLY, LDLRAD4, SUB1, PHYKPL 
## Negative:  FCGR3B, S100A9, SOD2, MNDA, SERPINA1, S100A8, IL8, LST1, FTH1, CXCR2 
##     LRRK2, FPR1, G0S2, CMTM2, NCF2, BCL2A1, TNFAIP2, GCA, PTGS2, AMICA1 
##     S100P, IER3, CD55, SPI1, SELL, FFAR2, EGR1, GLUL, LINC00936, STEAP4 
## PC_ 4 
## Positive:  NUPR1, RNASE1, SPP1, CPA1, CLPS, PNLIP, CELA3A, CTRC, PLA2G1B, CPB1 
##     REG1A, CELA2A, SPINK1, CELA3B, CTRB1, AMY2A, CPA2, CELA2B, SYCN, PRSS1 
##     GP2, CEL, PRSS3, CTRB2, KLK1, AMY2B, PNLIPRP1, GPNMB, APOE, APOC1 
## Negative:  FCN1, RP11-1143G9.4, CLEC12A, LGALS2, CCR2, F13A1, CD1D, MPEG1, MS4A6A, POU2F2 
##     CFP, S100A12, ASGR1, CD300E, NRGN, S100A4, PLBD1, CPVL, CD302, CSTA 
##     CYBB, AP1S2, C19orf38, LYZ, SLC46A2, ASGR2, ARL5B, GPBAR1, CD36, FGL2 
## PC_ 5 
## Positive:  S100A11, FCGR3B, IL8, CTSD, RARRES1, KRT19, S100P, FTH1, C15orf48, SPP1 
##     G0S2, S100A14, SMIM22, CCR1, ISG15, SOD2, ERO1L, PLA2G16, C19orf33, TYMP 
##     ZFP36L1, WFDC2, KRT7, MX1, APOL1, IFI6, EGLN3, AQP9, CXCL16, BCL2A1 
## Negative:  C11orf96, BGN, EDNRA, SNHG18, TBX2, DCN, PCDH18, CDH11, COL4A2, IGFBP7 
##     SPARC, COX7A1, COL1A1, COL3A1, LUM, PCOLCE, TAGLN, EMILIN1, PRELP, PKIG 
##     COL1A2, SYNPO2, AEBP1, SEPT4, GNG11, SOD3, PDGFRB, CALD1, GUCY1A3, PTK7</code></pre>
<pre class="r"><code>print(pdac.seurat[[&quot;pca&quot;]], dims = 1:5, nfeatures = 5)</code></pre>
<pre><code>## PC_ 1 
## Positive:  CTSB, CD68, TGFBI, GPNMB, MAFB 
## Negative:  CD7, CD247, CTSW, PRF1, CCL5 
## PC_ 2 
## Positive:  SMIM22, C19orf33, GPRC5A, PMEPA1, PDZK1IP1 
## Negative:  FTL, AIF1, NCF2, S100A8, LST1 
## PC_ 3 
## Positive:  CD7, CTSW, GZMA, CCND2, CCL5 
## Negative:  FCGR3B, S100A9, SOD2, MNDA, SERPINA1 
## PC_ 4 
## Positive:  NUPR1, RNASE1, SPP1, CPA1, CLPS 
## Negative:  FCN1, RP11-1143G9.4, CLEC12A, LGALS2, CCR2 
## PC_ 5 
## Positive:  S100A11, FCGR3B, IL8, CTSD, RARRES1 
## Negative:  C11orf96, BGN, EDNRA, SNHG18, TBX2</code></pre>
<pre class="r"><code>VizDimLoadings(pdac.seurat, dims = 1:2, reduction = &quot;pca&quot;)</code></pre>
<p><img src="TD_scRNA-seq_Seurat_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<pre class="r"><code>DimPlot(pdac.seurat, reduction = &quot;pca&quot;)</code></pre>
<p><img src="TD_scRNA-seq_Seurat_files/figure-html/unnamed-chunk-8-2.png" width="672" /></p>
<pre class="r"><code>DimHeatmap(pdac.seurat, dims = 1:10, cells = 500, balanced = TRUE)</code></pre>
<p><img src="TD_scRNA-seq_Seurat_files/figure-html/unnamed-chunk-8-3.png" width="672" /></p>
<pre class="r"><code>pdac.seurat</code></pre>
<pre><code>## An object of class Seurat 
## 19273 features across 928 samples within 1 assay 
## Active assay: RNA (19273 features, 2000 variable features)
##  3 layers present: counts, data, scale.data
##  1 dimensional reduction calculated: pca</code></pre>
</div>
<div id="estimation-de-la-dimensionalité" class="section level3"
number="3.6.2">
<h3><span class="header-section-number">3.6.2</span> Estimation de la
Dimensionalité</h3>
<p>Nous utilisons la méthode des valeurs propres obtenues à partir de
l’ACP.</p>
<ol style="list-style-type: decimal">
<li><strong>Faire un graphe</strong> des valeurs propres</li>
<li><strong>Estimer et discuter</strong> de la dimensionalité</li>
</ol>
<pre class="r"><code>ElbowPlot(pdac.seurat)</code></pre>
<p><img src="TD_scRNA-seq_Seurat_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
</div>
</div>
<div id="clusteriser-les-cellules" class="section level2" number="3.7">
<h2><span class="header-section-number">3.7</span> Clusteriser les
cellules</h2>
<ol style="list-style-type: decimal">
<li><strong>Trouver les plus proches voisins</strong> en construisant un
graphe <em>K-nearest neighbors</em> à l’aide de la fonction
<code>FindNeighbors</code></li>
<li><strong>Clusteriser les cellules</strong> en utilisant l’algorithme
de Louvain pour grouper les cellules (fonction
<code>FindClusters</code>)</li>
<li>Examiner les résultats pour les 5 premières cellules (utiliser
<code>Idents(pdac.seurat)</code>)</li>
</ol>
<div id="solution-3" class="section level3" number="3.7.1">
<h3><span class="header-section-number">3.7.1</span> Solution</h3>
<pre class="r"><code>pdac.seurat &lt;- FindNeighbors(pdac.seurat, reduction = &quot;pca&quot;, dims = 1:15)</code></pre>
<pre><code>## Computing nearest neighbor graph</code></pre>
<pre><code>## Computing SNN</code></pre>
<pre class="r"><code>pdac.seurat &lt;- FindClusters(pdac.seurat, resolution = 0.5)</code></pre>
<pre><code>## Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck
## 
## Number of nodes: 928
## Number of edges: 25631
## 
## Running Louvain algorithm...
## Maximum modularity in 10 random starts: 0.9121
## Number of communities: 10
## Elapsed time: 0 seconds</code></pre>
<pre class="r"><code>head(Idents(pdac.seurat), 5)</code></pre>
<pre><code>## AAACGAAAGTGGAAAG-1 AAAGAACCATTAAAGG-1 AAAGGATTCGGCTTGG-1 AAAGGGCAGTAGCAAT-1 
##                  0                  3                  1                  3 
## AAAGGTACACATACGT-1 
##                  0 
## Levels: 0 1 2 3 4 5 6 7 8 9</code></pre>
</div>
</div>
<div id="exploration-des-clusters-par-umap" class="section level2"
number="3.8">
<h2><span class="header-section-number">3.8</span> Exploration des
clusters par UMAP</h2>
<p>Seurat propose plusieurs techniques de réduction de dimension non
linéaires, telles que <strong>t-SNE</strong> et <strong>UMAP</strong>,
pour visualiser et explorer ces jeux de données. L’objectif de ces
algorithmes est d’identifier la <strong>structure sous-jacente</strong>
du jeu de données, afin de <strong>positionner des cellules similaires
les unes près des autres</strong> dans un espace de faible dimension.
Ainsi, les cellules regroupées dans les clusters définis par les
méthodes de regroupement basées sur les graphes devraient “<strong>se
retrouver proches</strong>” les unes des autres sur ces représentations
réduites.</p>
<ol style="list-style-type: decimal">
<li>Utiliser <code>RunUMAP</code> sur 15 dimensions puis
<code>Dimplot</code></li>
<li>Que pensez vous du nombre de clusters identifiés ?</li>
<li>Sauver l’objet <code>pdac.seurat</code>dans un fichier RDS pour
pouvoir le réutiliser plus tard.</li>
</ol>
<div id="solution-4" class="section level3" number="3.8.1">
<h3><span class="header-section-number">3.8.1</span> Solution</h3>
<pre class="r"><code>pdac.seurat &lt;- RunUMAP(pdac.seurat, dims = 1:15)</code></pre>
<pre><code>## 20:27:33 UMAP embedding parameters a = 0.9922 b = 1.112</code></pre>
<pre><code>## 20:27:33 Read 928 rows and found 15 numeric columns</code></pre>
<pre><code>## 20:27:33 Using Annoy for neighbor search, n_neighbors = 30</code></pre>
<pre><code>## 20:27:33 Building Annoy index with metric = cosine, n_trees = 50</code></pre>
<pre><code>## 0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
<pre><code>## [----|----|----|----|----|----|----|----|----|----|</code></pre>
<pre><code>## **************************************************|
## 20:27:33 Writing NN index file to temp file /var/folders/91/6skgr68s5ls_dhqfdzs02_pm00013r/T//RtmpYvXvhN/filea93748acb073
## 20:27:33 Searching Annoy index using 1 thread, search_k = 3000
## 20:27:33 Annoy recall = 100%
## 20:27:33 Commencing smooth kNN distance calibration using 1 thread with target n_neighbors = 30
## 20:27:34 Initializing from normalized Laplacian + noise (using RSpectra)
## 20:27:34 Commencing optimization for 500 epochs, with 34120 positive edges
## 20:27:34 Using rng type: pcg
## 20:27:35 Optimization finished</code></pre>
<pre class="r"><code>DimPlot(pdac.seurat, reduction = &quot;umap&quot;)</code></pre>
<p><img src="TD_scRNA-seq_Seurat_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<pre class="r"><code>saveRDS(pdac.seurat, file = &quot;pdac_1.rds&quot;)

# Alternative: SaveSeuratRds()

pdac.seurat</code></pre>
<pre><code>## An object of class Seurat 
## 19273 features across 928 samples within 1 assay 
## Active assay: RNA (19273 features, 2000 variable features)
##  3 layers present: counts, data, scale.data
##  2 dimensional reductions calculated: pca, umap</code></pre>
</div>
</div>
<div id="visualisation-des-marqueurs" class="section level2"
number="3.9">
<h2><span class="header-section-number">3.9</span> Visualisation des
marqueurs</h2>
<p>Cette étape permet de découvrir les types cellulaires les plus
connus. Plusieurs outils de visualisation sont pertinents pour cela: Les
<em>Violin plots</em> (fonction <code>VlnPlot</code>) et
<strong>Superposition de l’expression d’un marqueur</strong> sur l’UMAP
(Fonction <code>Featureplot</code>).</p>
<ol style="list-style-type: decimal">
<li>Mettez en avant les <strong>cellules immunitaires</strong> par la
visualisation du marqueur <em>CD45</em></li>
<li>Mettez en avant les types cellulaires suivants, caractérisés par les
marqeurs suivants:</li>
</ol>
<table>
<thead>
<tr class="header">
<th>Type cellulaire</th>
<th>Marqueur</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Dendritic cells</td>
<td>CLU, PRSS1, HLA-DRA</td>
</tr>
<tr class="even">
<td>B Cells</td>
<td>HLA-DRA, CD79A, MS4A1</td>
</tr>
<tr class="odd">
<td>Macrophages</td>
<td>CD14, HLA-DRA, LYZ</td>
</tr>
<tr class="even">
<td>CD8+ T cells</td>
<td>CD3E, NKG7, CD3D</td>
</tr>
<tr class="odd">
<td>CD4+ T cells</td>
<td>CD3E, CD3D</td>
</tr>
<tr class="even">
<td>Epithelial cells</td>
<td>KRT18, KRT19,</td>
</tr>
<tr class="odd">
<td>NK cells</td>
<td>NKG7, GZMB</td>
</tr>
</tbody>
</table>
<ol start="3" style="list-style-type: decimal">
<li>Faites un <code>Dotplot</code> avec l’ensemble de ces marqueurs</li>
</ol>
<pre class="r"><code># Plots pour CD45
VlnPlot(pdac.seurat, features = &quot;PTPRC&quot;)</code></pre>
<p><img src="TD_scRNA-seq_Seurat_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<pre class="r"><code>FeaturePlot(pdac.seurat, features = &quot;PTPRC&quot;)</code></pre>
<p><img src="TD_scRNA-seq_Seurat_files/figure-html/unnamed-chunk-12-2.png" width="672" /></p>
<pre class="r"><code># On peut aussi mettre en place une analyse interactive
#FeaturePlot(pdac.seurat, features = &quot;PTPRC&quot;, interactive = TRUE)

# Plots pour notre liste de marqeurs
genes_markers &lt;- c(
  &quot;CLU&quot;, &quot;PRSS1&quot;, &quot;HLA-DRA&quot;,
  &quot;CD79A&quot;, &quot;MS4A1&quot;,
  &quot;CD14&quot;, &quot;LYZ&quot;,
  &quot;CD3E&quot;, &quot;NKG7&quot;, &quot;CD3D&quot;,
  &quot;KRT18&quot;, &quot;KRT19&quot;,
  &quot;GZMB&quot;
)

VlnPlot(pdac.seurat, features = genes_markers)</code></pre>
<p><img src="TD_scRNA-seq_Seurat_files/figure-html/unnamed-chunk-12-3.png" width="672" /></p>
<pre class="r"><code>FeaturePlot(pdac.seurat, features = genes_markers)</code></pre>
<p><img src="TD_scRNA-seq_Seurat_files/figure-html/unnamed-chunk-12-4.png" width="672" /></p>
<pre class="r"><code>dtplot &lt;- DotPlot(pdac.seurat, features = genes_markers) + RotatedAxis()
plot(dtplot)</code></pre>
<p><img src="TD_scRNA-seq_Seurat_files/figure-html/unnamed-chunk-12-5.png" width="672" /></p>
</div>
<div id="identifier-les-marqueurs-différentiellement-exprimés"
class="section level2" number="3.10">
<h2><span class="header-section-number">3.10</span> Identifier les
marqueurs différentiellement exprimés</h2>
<ol style="list-style-type: decimal">
<li>Créer une table avec la liste des marquers
<strong>différentiellement exprimés</strong> (DE) entre le cluster 2
(désigné sous le nom de <code>ìdent.1 = 2</code>) et le reste des
clusters.</li>
<li>afficher les <strong>10 premiers</strong> marqueurs</li>
<li>Identifier les marquers DE entre le <strong>cluster 2</strong> et le
<strong>cluster 3</strong> et afficher les <strong>10</strong> premiers
marqueurs.</li>
<li>Identifier les marqueurs de différentiation pour l’ensemble des
clusters en utilisant <code>FindAllMarkers</code> puis filtrer les
marqueurs sur <span class="math inline">\(avglog2FC &gt; 2.0\)</span> en
utilisant la librairie <code>dplyr</code>.</li>
</ol>
<pre class="r"><code># On vérifie que l&#39;on est bien sur le bon Ident
table(Idents(pdac.seurat))</code></pre>
<pre><code>## 
##   0   1   2   3   4   5   6   7   8   9 
## 194 188 112  93  81  81  62  55  32  30</code></pre>
<pre class="r"><code># 1. Table avec la liste des marqueurs DE
cluster2.markers &lt;- FindMarkers(pdac.seurat, ident.1 = 2)</code></pre>
<pre><code>## For a (much!) faster implementation of the Wilcoxon Rank Sum Test,
## (default method for FindMarkers) please install the presto package
## --------------------------------------------
## install.packages(&#39;devtools&#39;)
## devtools::install_github(&#39;immunogenomics/presto&#39;)
## --------------------------------------------
## After installation of presto, Seurat will automatically use the more 
## efficient implementation (no further action necessary).
## This message will be shown once per session</code></pre>
<pre class="r"><code># 2. afficher les 10 premiers marqueurs
head(cluster2.markers, n = 10)</code></pre>
<pre><code>##                   p_val avg_log2FC pct.1 pct.2     p_val_adj
## MAL       2.901016e-120   9.118145 0.643 0.004 5.591127e-116
## IL7R      1.007071e-117   4.380175 0.920 0.065 1.940928e-113
## LEF1      1.865256e-114   6.892020 0.643 0.007 3.594907e-110
## CCR7      9.776392e-107   5.850119 0.705 0.026 1.884204e-102
## CD3G      2.724034e-103   3.019207 0.946 0.078  5.250032e-99
## CD3D       1.167385e-90   2.727310 0.938 0.100  2.249901e-86
## TCF7       6.045381e-84   4.357094 0.812 0.102  1.165126e-79
## CD27       8.387488e-79   3.082440 0.732 0.062  1.616520e-74
## PRKCQ-AS1  1.006525e-72   4.120219 0.723 0.085  1.939876e-68
## BCL11B     1.472182e-71   3.818475 0.652 0.056  2.837337e-67</code></pre>
<pre class="r"><code># 3. pareil pour les cluster 2 et cluster s3 
cluster2.3.markers &lt;- FindMarkers(pdac.seurat, ident.1 = 2, ident.2 = 3)
head (cluster2.3.markers, n = 10)</code></pre>
<pre><code>##                 p_val avg_log2FC pct.1 pct.2    p_val_adj
## GPNMB    5.341285e-40  -8.552400 0.027 0.989 1.029426e-35
## CD68     8.306483e-40  -9.339173 0.009 0.978 1.600909e-35
## LYZ      1.674663e-39  -9.233372 0.018 0.978 3.227578e-35
## PLAUR    1.723876e-39  -6.612930 0.071 1.000 3.322426e-35
## C15orf48 6.927291e-39  -7.881700 0.018 0.968 1.335097e-34
## FCER1G   2.039545e-38  -6.028574 0.036 0.978 3.930816e-34
## GNS      2.607417e-38  -6.402631 0.036 0.968 5.025276e-34
## CTSL     3.686963e-38  -7.463309 0.152 1.000 7.105883e-34
## TYROBP   5.626096e-38  -6.928897 0.054 0.968 1.084317e-33
## TIMP2    5.816277e-38  -6.453160 0.054 0.968 1.120971e-33</code></pre>
<pre class="r"><code># 4. marqueurs pour l&#39;ensemble des clusters
pdac.all.markers &lt;- FindAllMarkers(pdac.seurat, only.pos = TRUE)</code></pre>
<pre><code>## Calculating cluster 0</code></pre>
<pre><code>## Calculating cluster 1</code></pre>
<pre><code>## Calculating cluster 2</code></pre>
<pre><code>## Calculating cluster 3</code></pre>
<pre><code>## Calculating cluster 4</code></pre>
<pre><code>## Calculating cluster 5</code></pre>
<pre><code>## Calculating cluster 6</code></pre>
<pre><code>## Calculating cluster 7</code></pre>
<pre><code>## Calculating cluster 8</code></pre>
<pre><code>## Calculating cluster 9</code></pre>
<pre class="r"><code># Filtrage du resultat (logFC&gt;1) et tri par cluster
library(dplyr)</code></pre>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>pdac.all.markers %&gt;% dplyr::group_by(cluster) %&gt;% dplyr::filter(avg_log2FC &gt; 2.0) -&gt; filtered_markers</code></pre>
</div>
<div id="représentation-de-lexpression-sous-forme-de-heatmap"
class="section level2" number="3.11">
<h2><span class="header-section-number">3.11</span> Représentation de
l’expression sous forme de heatmap</h2>
<ol style="list-style-type: decimal">
<li>Faire une <strong>représentation heatmap</strong> des gènes de notre
liste <code>genes_markers</code> (fonction <code>DoHeatmap</code>)</li>
<li>Faire une représentation des <strong>10 gènes les plus DE</strong>
caractérisant chaque cluster (appliquer les fonctions
<code>group_by</code>, <code>filter</code>, <code>slice_head</code> et
<code>ungroup</code> du package <code>dplyr</code>)</li>
</ol>
<div id="solution-5" class="section level3" number="3.11.1">
<h3><span class="header-section-number">3.11.1</span> Solution</h3>
<pre class="r"><code># heatmap liste marqueurs
DoHeatmap(pdac.seurat, features = genes_markers) + NoLegend()</code></pre>
<p><img src="TD_scRNA-seq_Seurat_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<pre class="r"><code># heatmap du top10 genes DE pout tous les clusters
library(dplyr)
pdac.all.markers %&gt;%
    dplyr::group_by(cluster) %&gt;%
    dplyr::filter(avg_log2FC &gt; 1) %&gt;%
    dplyr::slice_head(n = 10) %&gt;%
    dplyr::ungroup() -&gt; top10

DoHeatmap(pdac.seurat, features = top10$gene) + NoLegend()</code></pre>
<p><img src="TD_scRNA-seq_Seurat_files/figure-html/unnamed-chunk-14-2.png" width="672" /></p>
</div>
</div>
<div id="exemple-dutilisation-de-addmodulescore" class="section level2"
number="3.12">
<h2><span class="header-section-number">3.12</span> Exemple
d’utilisation de <code>AddmoduleScore</code></h2>
<p>Calcul du score de la signature ‘TOP10’</p>
<pre class="r"><code>activation_genes &lt;- top10

pdac.seurat &lt;- AddModuleScore(
  pdac.seurat,
  features = list(activation_genes),
  name = &quot;TActivation&quot;
)</code></pre>
<p>Plot du score</p>
<pre class="r"><code>FeaturePlot(
  pdac.seurat,
  features = &quot;TActivation1&quot;,
  cols = c(&quot;lightgrey&quot;, &quot;red&quot;),
  reduction = &quot;umap&quot;
)</code></pre>
<p><img src="TD_scRNA-seq_Seurat_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
</div>
</div>
<div id="session-info" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Session Info</h1>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>## R version 4.5.1 (2025-06-13)
## Platform: aarch64-apple-darwin20
## Running under: macOS Sequoia 15.7.1
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRblas.0.dylib 
## LAPACK: /Library/Frameworks/R.framework/Versions/4.5-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.1
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## time zone: Europe/Paris
## tzcode source: internal
## 
## attached base packages:
## [1] stats     graphics  grDevices datasets  utils     methods   base     
## 
## other attached packages:
## [1] dplyr_1.1.4        future_1.67.0      Seurat_5.3.1       SeuratObject_5.2.0
## [5] sp_2.2-0          
## 
## loaded via a namespace (and not attached):
##   [1] deldir_2.0-4           pbapply_1.7-4          gridExtra_2.3         
##   [4] rlang_1.1.6            magrittr_2.0.4         RcppAnnoy_0.0.22      
##   [7] otel_0.2.0             spatstat.geom_3.6-0    matrixStats_1.5.0     
##  [10] ggridges_0.5.7         compiler_4.5.1         png_0.1-8             
##  [13] vctrs_0.6.5            reshape2_1.4.4         stringr_1.5.2         
##  [16] pkgconfig_2.0.3        fastmap_1.2.0          labeling_0.4.3        
##  [19] promises_1.5.0         rmarkdown_2.30         purrr_1.1.0           
##  [22] xfun_0.54              cachem_1.1.0           jsonlite_2.0.0        
##  [25] goftest_1.2-3          later_1.4.4            spatstat.utils_3.2-0  
##  [28] irlba_2.3.5.1          parallel_4.5.1         cluster_2.1.8.1       
##  [31] R6_2.6.1               ica_1.0-3              spatstat.data_3.1-9   
##  [34] bslib_0.9.0            stringi_1.8.7          RColorBrewer_1.1-3    
##  [37] limma_3.64.3           reticulate_1.44.0      spatstat.univar_3.1-4 
##  [40] parallelly_1.45.1      lmtest_0.9-40          jquerylib_0.1.4       
##  [43] scattermore_1.2        Rcpp_1.1.0             knitr_1.50            
##  [46] tensor_1.5.1           future.apply_1.20.0    zoo_1.8-14            
##  [49] R.utils_2.13.0         sctransform_0.4.2      httpuv_1.6.16         
##  [52] Matrix_1.7-4           splines_4.5.1          igraph_2.2.1          
##  [55] tidyselect_1.2.1       abind_1.4-8            rstudioapi_0.17.1     
##  [58] yaml_2.3.10            spatstat.random_3.4-2  codetools_0.2-20      
##  [61] miniUI_0.1.2           spatstat.explore_3.5-3 listenv_0.10.0        
##  [64] lattice_0.22-7         tibble_3.3.0           plyr_1.8.9            
##  [67] withr_3.0.2            shiny_1.11.1           S7_0.2.0              
##  [70] ROCR_1.0-11            evaluate_1.0.5         Rtsne_0.17            
##  [73] fastDummies_1.7.5      survival_3.8-3         polyclip_1.10-7       
##  [76] fitdistrplus_1.2-4     pillar_1.11.1          BiocManager_1.30.26   
##  [79] KernSmooth_2.23-26     renv_1.1.5             plotly_4.11.0         
##  [82] generics_0.1.4         RcppHNSW_0.6.0         ggplot2_4.0.0         
##  [85] scales_1.4.0           globals_0.18.0         xtable_1.8-4          
##  [88] glue_1.8.0             lazyeval_0.2.2         tools_4.5.1           
##  [91] data.table_1.17.8      RSpectra_0.16-2        RANN_2.6.2            
##  [94] dotCall64_1.2          cowplot_1.2.0          grid_4.5.1            
##  [97] tidyr_1.3.1            nlme_3.1-168           patchwork_1.3.2       
## [100] cli_3.6.5              spatstat.sparse_3.1-0  spam_2.11-1           
## [103] viridisLite_0.4.2      uwot_0.2.3             gtable_0.3.6          
## [106] R.methodsS3_1.8.2      sass_0.4.10            digest_0.6.37         
## [109] progressr_0.17.0       ggrepel_0.9.6          htmlwidgets_1.6.4     
## [112] farver_2.1.2           R.oo_1.27.1            htmltools_0.5.8.1     
## [115] lifecycle_1.0.4        httr_1.4.7             statmod_1.5.1         
## [118] mime_0.13              MASS_7.3-65</code></pre>
<hr />
<div style="text-align: center">
<p><a rel="license" href="http://creativecommons.org/licenses/by-nc-nd/4.0/"><img src="https://licensebuttons.net/l/by-nc-nd/4.0/88x31.png" alt="Licence Creative Commons" style="border-width:0"/></a><br></p>
<p>Cette œuvre est mise à disposition selon les termes de la
<a rel="license" href="http://creativecommons.org/licenses/by-nc-nd/4.0/">Licence
Creative Commons:</a> <br>Attribution - Pas d’Utilisation Commerciale -
Pas de Modification 4.0 International (CC BY-NC-ND 4.0).</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
